nixos/tests/php/unit.nix

```diff
@@ -0,0 +1,101 @@
+import ../make-test-python.nix ({pkgs, ...}:
+ let
+    testdir = pkgs.writeTextDir "www/info.php" ''
+      <?php
+        phpinfo();
+      ?>
+    '';
+
+in {
+  name = "unit-php-test";
+  meta.maintainers = with pkgs.stdenv.lib.maintainers; [ izorkin ];
+
+  machine = { config, lib, pkgs, ... }: {
+    nixpkgs.overlays = [
+      (self: super: {
+        unit-test = pkgs.unit.override {
+          withPHP72 = true;
+          withPHP73 = true;
+          withPHP74 = true;
+        };
+      })
+    ];
+    services.unit = {
+      enable = true;
+      package = pkgs.unit-test;
+      config = ''
+        {
+          "listeners": {
+            "*:9002": {
+              "application": "php_72"
+            },
+            "*:9003": {
+              "application": "php_73"
+            },
+            "*:9004": {
+              "application": "php_74"
+            }
+          },
+          "applications": {
+            "php_72": {
+              "type": "php 7.2",
+              "processes": 1,
+              "user": "testuser2",
+              "group": "testgroup2",
+              "root": "${testdir}/www",
+              "index": "info.php"
+            },
+            "php_73": {
+              "type": "php 7.3",
+              "processes": 1,
+              "user": "testuser3",
+              "group": "testgroup3",
+              "root": "${testdir}/www",
+              "index": "info.php"
+            },
+            "php_74": {
+              "type": "php 7.4",
+              "processes": 1,
+              "user": "testuser4",
+              "group": "testgroup4",
+              "root": "${testdir}/www",
+              "index": "info.php"
+            }
+          }
+        }
+      '';
+    };
+    users = {
+      users.testuser2 = {
+        isNormalUser = false;
+        uid = 1002;
+        group = "testgroup2";
+      };
+      users.testuser3 = {
+        isNormalUser = false;
+        uid = 1003;
+        group = "testgroup3";
+      };
+      users.testuser4 = {
+        isNormalUser = false;
+        uid = 1004;
+        group = "testgroup4";
+      };
+      groups.testgroup2 = {
+        gid= 1002;
+      };
+      groups.testgroup3 = {
+        gid= 1003;
+      };
+      groups.testgroup4 = {
+        gid= 1004;
+      };
+    };
+  };
+  testScript = ''
+    machine.wait_for_unit("unit.service")
+    assert "PHP Version ${pkgs.php72.version}" in machine.succeed("curl -vvv -s http://127.0.0.1:9002/")
+    assert "PHP Version ${pkgs.php73.version}" in machine.succeed("curl -vvv -s http://127.0.0.1:9003/")
+    assert "PHP Version ${pkgs.php74.version}" in machine.succeed("curl -vvv -s http://127.0.0.1:9004/")
```

Comment:
Maybe these asserts should be a bit more extensive, such as looking for enabled extensions as we do in `fpm.nix` and such?

Reply:
Php extensions not builded in unit :(
https://github.com/NixOS/nixpkgs/pull/85026#issuecomment-614761982